src/modules/commander/Commander.cpp

```diff
@@ -1737,7 +1737,7 @@ Commander::run()
 
 		/* update subsystem info which arrives from outside of commander*/
 		do {
-			if (subsys_sub.updated()) {
+			if ( (updated=subsys_sub.updated()) ) {
```

Comment:
Wow,  this `updated` bool is defined 500 lines above. It looks like it's not used anywhere else. Could you move it down to here where it's used?

Reply:
Actually, can't you just simplify it like this?

```
		/* update subsystem info which arrives from outside of commander*/
		while (subsys_sub.updated()) {
			subsystem_info_s info{};
			subsys_sub.copy(&info);
			set_health_flags(info.subsystem_type, info.present, info.enabled, info.ok, status);
			status_changed = true;
		}
```

Reply:
yes, changed

Reply:
also tested, it gets at most 5 messages in 1 commander loop, which is limited by the messages orb queue size